### DISCUSSION TITLE
"Click" for menubar item?

### DISCUSSION THREAD

**Discussion by GronbergInfo** [tznind/TerminalGuiDesigner#207] [General]
hi, i have been looking around and cannot seem to fine a way to do something when a menu item is clicked.
Is it action i need to use? If so, got any example?

**Comment by tznind** [comment]
Currently the designer does not support creating events but you can do it with the `Action` property on the MenuItem:

For example if you have an empty view that you have added a MenuBar to there will be a placeholder menu item called "Edit Me".  Putting a callback on that is done as follows (In MyView.cs):

```csharp
public MyView() {
    InitializeComponent();

    editMeMenuItem.Action = SomeMethod;
}

private void SomeMethod()
{
    MessageBox.Query("Clicked", "You clicked bud!", "Yeah!");
}
```

**Reply by GronbergInfo** [reply]
Thank you, got it working as i wished for.
